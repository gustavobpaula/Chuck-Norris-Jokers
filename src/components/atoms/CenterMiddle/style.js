import styled from 'styled-components';

export const Center = styled.div`
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;
