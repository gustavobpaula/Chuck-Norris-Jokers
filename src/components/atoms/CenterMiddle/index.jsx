import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './style';

const CenterMiddle = ({ children }) => (
  <Styled.Center>{children}</Styled.Center>
);

CenterMiddle.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CenterMiddle;
