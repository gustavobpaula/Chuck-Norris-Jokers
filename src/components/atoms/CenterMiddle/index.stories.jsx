import React from 'react';
import CenterMiddle from './index';

export default {
  component: CenterMiddle,
  title: 'Atoms/CenterMiddle',
};

export const element = () => (
  <CenterMiddle>
    <h1>Centered Content</h1>
  </CenterMiddle>
);
element.story = {
  name: 'Default',
};
