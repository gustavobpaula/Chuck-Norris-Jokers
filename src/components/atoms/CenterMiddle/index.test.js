import React from 'react';
import { mount } from 'enzyme';

import CenterMiddle from './index';

describe('<CenterMiddle />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(
      <CenterMiddle>
        <h1>Centered Content</h1>
      </CenterMiddle>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
