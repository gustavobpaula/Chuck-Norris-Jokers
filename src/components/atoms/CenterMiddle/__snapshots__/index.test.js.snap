// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterMiddle /> should mount properly 1`] = `
<CenterMiddle>
  <style__Center>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Center-sc-1qm54d6-0",
            "isStatic": false,
            "lastClassName": "hWvMyz",
            "rules": Array [
              "left:50%;position:absolute;top:50%;transform:translate(-50%,-50%);",
            ],
          },
          "displayName": "style__Center",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Center-sc-1qm54d6-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="style__Center-sc-1qm54d6-0 hWvMyz"
      >
        <h1>
          Centered Content
        </h1>
      </div>
    </StyledComponent>
  </style__Center>
</CenterMiddle>
`;
